core/thread/name_spec.rb

```diff
@@ -0,0 +1,22 @@
+require File.expand_path('../../../spec_helper', __FILE__)
+require File.expand_path('../fixtures/classes', __FILE__)
+
+ruby_version_is '2.3' do
+  describe "Thread#name" do
+    before do
+      @thread = Thread.new { 3 }
+      @thread.name = :thread_name
+    end
+
```

Comment:
I'd be interested in what is the initial value of name, and its class.
The value after the setter can be tested only in Thread#name= to avoid duplication.